storage_v2/factoryadapter/reader.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package factoryadapter
+
+import (
+	"context"
+	"errors"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	spanstore_v1 "github.com/jaegertracing/jaeger/storage/spanstore"
+	"github.com/jaegertracing/jaeger/storage_v2/spanstore"
+)
+
+var ErrV1ReaderNotAvailable = errors.New("v1 reader is not available")
```

Comment:
I am not yet seeing a need for this being a public variable
```suggestion
var errV1ReaderNotAvailable = errors.New("spanstore.Reader is not a wrapper around v1 reader")
```